WooCommerce/Classes/ViewRelated/Products/Edit Product/Linked Products/LinkedProductsViewModel.swift

```diff
@@ -0,0 +1,76 @@
+import UIKit
+import Yosemite
+
+/// Provides data needed for Linked Products.
+///
+protocol LinkedProductsViewModelOutput {
```

Comment:
I'm just curious here: what is the reason behind the two protocols? I understand this provides a kind of [command-query separation](https://martinfowler.com/bliki/CommandQuerySeparation.html) but I don't see the practical benefit of formalizing that separation. Wha am I missing?

Leaving that aside, there is one more thing I am curious about: conformance to LinkedProductsViewModelActionHandler is implemented in an extension, while conformance to LinkedProductsViewModelOutput is implemented in the class itself. That reads a bit un-symmetric, would it make sense to implement conformance to each protocol in a separate extension?

Reply:
Probably you are right, declaring two protocols for formalizing a separation is a little bit overkill here (also because in the future this view model will not grow), so I removed the two protocols 👍 5cfc7159a499e53bb55cceff22401fb6e1e8cbd4